lessons/intro/pathlib/index.md

```diff
@@ -0,0 +1,621 @@
+
+```pycon
+>>> # Existuje na té ukázkové cestě nějaký soubor?
+>>> cesta.exists()
+False
+
+>>> # Existuje můj domovský adresář?
+>>> Path.home().exists()
+True
+
+>>> # A je to vůbec adresář?
+>>> Path.home().is_dir()
+True
+
+>>> # Je to normální datový soubor?
+>>> Path.home().is_file()
+False
+```
+
+
+## Ukázka
+
+Abychom měli všichni stejné podmínky, stáhni si na další experimenty
+[archiv s testovacími soubory](static/archiv.tar.gz).
+Dej si ho do aktuálního adresáře (`Path.cwd()`), a pak ho rozbal pomocí
+`tarfile`:
+
+```pycon
+>>> import tarfile
+
+>>> cesta_k_archivu = Path("archiv.tar.gz")
+
+>>> # Co je v archivu?
+>>> tarfile.open(cesta_k_archivu, 'r|gz').getnames()
+['soubory',
+ 'soubory/hrad.jpeg',
+ 'soubory/hrad.attribution',
+ 'soubory/.gitignore',
+ 'soubory/kolecko.png',
+ 'soubory/texty',
+ 'soubory/texty/vodnik.txt',
+ 'soubory/texty/lidove',
+ 'soubory/texty/lidove/pes.txt',
+ 'soubory/texty/lidove/holka.txt',
+ 'soubory/texty/vladimir.txt',
+ 'soubory/texty/cizojazycne',
+ 'soubory/texty/cizojazycne/iroha.txt',
+ 'soubory/texty/cizojazycne/witch.txt',
+ 'soubory/hlad.txt',
+ 'soubory/hraz.attribution',
+ 'soubory/ententyky.txt',
+ 'soubory/hraz.jpeg',
+ 'soubory/README']
+
+>>> # Extrakce archivu. (Kdybys to zkoušel/a pro jiné archivy, vždy před
+>>> # rozbalením zkontroluj cesty všech souborů v archivu -- ať se rozbalením
+>>> # nepřepíše nějaký důležitý soubor!)
+>>> tarfile.open(cesta_k_archivu, 'r|gz').extractall()
+```
+
+Rozbalením archivu vznikl `./soubory/` (tedy: adresář `soubory` v aktuálním
+adresáři).
+Pojď se mu kouknout na zoubek:
+
+```pycon
+>>> zaklad = Path('./soubory')
+>>> zaklad
+PosixPath('soubory')
+
+>>> print('Je to adresář?', zaklad.is_dir())
+Je to adresář? True
+>>> print('Je to normální soubor?', zaklad.is_file())
+Je to normální soubor? False
+```
+
+Podle informací o archivu je v soubory nějaký `ententyky.txt` – podle přípony
+soubor s textem.
+
+```pycon
+>>> ententyky = zaklad / 'ententyky.txt'
+>>> print('Je to adresář?', ententyky.is_dir())
+Je to adresář? False
+>>> print('Je to normální soubor?', ententyky.is_file())
+Je to normální soubor? True
+```
+
+Objekty `Path` lze používat v naprosté většině situací, kdy jde použít cesta
+jako řetězec.
+Například pro funkci `open`:
+
+```python
+with open(ententyky, encoding='utf-8') as file:
+    print(file.read())
+```
+
+`Path` ale má `open` i jako metodu:
+
+```python
+with ententyky.open(encoding='utf-8') as file:
+    print(file.read())
+```
+
+A protože je čtení celého textového obsahu souboru docela užitečné,
+existuje i zkratka která soubor otevře, přečte a zavře najednou:
+
+```python
+print(ententyky.read_text())
+```
+
+(Větší soubory je ale lepší otevřít ve `with` a zpracovávat třeba po řádcích,
+aby se obsah nemusel do paměti počítače načíst celý najednou.)
+
+Existuje i `write_text`:
+
+```python
+cesta = Path.cwd() / 'pisnicka.txt'
+cesta.write_text('Holka modrooká\nNesedávej u potoka!')
+```
+
+{#
```

Comment:
Comment?

Reply:
Patří sem tahle část? Nepříjemně to navazuje na Notebook.

Reply:
To je sice fakt, ale zase není třeba hluboká znalost notebooku k pochopení problému. Bohužel mě nenapadá jiný příklad, kterým by to šlo nahradit.

Reply:
Možná by to šlo nahradit obecnou poznámkou bez konkrétního příkladu jen s ukázkou chybové hlášky, která je často jako přes kopírák.